libs/ui-lib/lib/ocm/components/clusterConfiguration/ExternalPlatformDropdown.tsx

```diff
@@ -0,0 +1,142 @@
+import {
+  NUTANIX_CONFIG_LINK,
+  PlatformType,
+  VSPHERE_CONFIG_LINK,
+  getFieldId,
+} from '../../../common';
+
+const INPUT_NAME = 'platform';
+const fieldId = getFieldId(INPUT_NAME, 'input');
+export type ExternalPlatformType = Extract<PlatformType, 'none' | 'nutanix' | 'oci' | 'vsphere'>;
+
+type ExternalPlatformDropdownProps = {
+  showOciOption: boolean;
+  selectedPlatform: ExternalPlatformType;
+  disabledOciTooltipContent: React.ReactNode;
+  isOciDisabled: boolean;
+  onChange: (selectedPlatform: ExternalPlatformType) => void;
+  dropdownIsDisabled: boolean;
+};
+
+export type ExternalPlatformInfo = {
+  label: string;
+  href: string;
+  tooltip: string;
+};
+
+export const externalPlatformTypes: Record<ExternalPlatformType, ExternalPlatformInfo> = {
+  none: {
+    label: 'None',
+    href: '',
+    tooltip: '',
+  },
+  nutanix: {
+    label: 'Nutanix',
+    href: NUTANIX_CONFIG_LINK,
+    tooltip: '',
+  },
+  oci: {
+    label: 'Oracle  (Requires a custom manifest)',
+    href: 'www.google.es',
+    tooltip:
+      "To integrate with an external partner (for example, Oracle Cloud), you'll need to provide a custom manifest.",
+  },
+  vsphere: {
+    label: 'vSphere',
+    href: VSPHERE_CONFIG_LINK,
+    tooltip: '',
+  },
+};
+
+export const ExternalPlatformDropdown = ({
+  showOciOption,
+  selectedPlatform,
+  disabledOciTooltipContent,
+  isOciDisabled,
+  onChange,
+  dropdownIsDisabled,
+}: ExternalPlatformDropdownProps) => {
+  const [field, { value }, { setValue }] = useField<string>(INPUT_NAME);
+  const [currentPlatform, setCurrentPlatform] = React.useState(
+    externalPlatformTypes[selectedPlatform].label,
+  );
+  const [isOpen, setOpen] = React.useState(false);
+  const enabledItems = Object.keys(externalPlatformTypes)
+    .filter((platformType) => {
+      if (platformType === 'oci') {
+        return showOciOption;
+      }
+      return true;
+    })
+    .map((platformType) => {
+      const { label, href, tooltip } = externalPlatformTypes[platformType as ExternalPlatformType];
+      const isOracleDisabled = platformType === 'oci' && isOciDisabled;
+      const isHrefEmpty = href === '';
+      return (
+        <>
+          <DropdownItem
+            key={platformType}
+            id={platformType}
+            tooltip={isOracleDisabled ? disabledOciTooltipContent : tooltip}
+            isAriaDisabled={isOracleDisabled}
+            tooltipProps={{ position: 'top-start' }}
+          >
+            {label}
+            {!isHrefEmpty && (
+              <a href={href} target="_blank" rel="noopener noreferrer" style={{ float: 'right' }}>
+                Learn more <i className="fas fa-external-link-alt" />
+              </a>
+            )}
+          </DropdownItem>
+        </>
+      );
+    });
+
+  const onSelect = React.useCallback(
+    (event?: React.SyntheticEvent<HTMLDivElement>) => {
+      const selectedPlatform = event?.currentTarget.id as ExternalPlatformType;
```

Comment:
It's possible to change the platform even when hosts are discovered and the platform does not match.
Shouldn't this be controlled, either on the UI or the Backend?

This is possible linked to the comments about `useClusterSupportedPlatforms` and the code in `HostDiscovery` that allowed the user to toggle on the switch and change the platform.

Reply:
@jkilzi we need to help to clarify this point

Reply:
https://redhat-internal.slack.com/archives/CUPJTHQ5P/p1689581850055749

Reply:
In this discussion @nirfarkas write this:
"  this drop-down should be agnostic to whatever we detected because the assumption is that users know what kind of hosts they booted/ gonna boot and also then we’ll need to explain why it’s disabled etc."